### ISSUE TITLE
NotABug: unable to create log: open: permissions denied

### ISSUE THREAD

**Issue by TekTimmy** [kubernetes/minikube#11205]
**Steps to reproduce the issue:** 

I want to use Minikube on a linux machine with several users, so this is what I did so far:

* Setup user for minikube and switch user
  ```bash
  adduser --disabled-login minikube
  usermod -aG docker minikube
  su minikube
  ```
* Start Minikube
  ```bash
  minikube config set vm-driver docker
  minikube config set cpus 4
  minikube config set memory 8192
  minikube config set disk-size 10G
  minikube config set kubernetes-version v1.18.17
  mkdir ~/logs
  minikube start --log_dir=~/logs
  ```
* Prepare access to Minikube to other users:
  ```bash
  chmod -R g+wrx ~/.minikube
  chmod -R g+r ~/.kube
  ```
* Provide access to Minikube to user "test-user":
  ```bash
  su test-user
  ln -s /home/minikube/.minikube ~/.minikube
  cp -r /home/minikube/.kube ~/.kube
  ```

```bash
~$ minikube version
minikube version: v1.19.0
commit: 15cede53bdc5fe242228853e737333b09d4336b5
```

When then one user1 runs `minikube ssh` and afterwards user2 does the same i get the following error message:
```
I0426 19:45:31.875059  366698 out.go:278] Setting OutFile to fd 1 ...
log: exiting because of error: log: unable to create log: open /tmp/minikube_ssh_6bca5e284ff1b988432c36e406f313bc2482ae72_0.log: permission denied
```

Even though I set the log_dir to another directory he is creating them in /tmp and why is the permission denied? All users are in group "minikube" so they should be able to overwrite that file...

**Comment by afbjorklund** [comment]
As long as we use `os.Args` for the hash, should probably throw the user name in there as well....

But another question is why the `log_dir` didn't work ? Seems like it is still using $TMP

**Comment by TekTimmy** [comment]
> As long as we use `os.Args` for the hash, should probably throw the user name in there as well....
Can i disable the logging? I don't think I need it 

> 
> But another question is why the `log_dir` didn't work ? Seems like it is still using ${TMPDIR:-/tmp}
Yes that might actually be a bug 🐛

**Comment by afbjorklund** [comment]
The log_file was different in older versions, and you might be able to use TMPDIR as a workaround

**Comment by TekTimmy** [comment]
Thank you @afbjorklund, works like a charm!
I have added the following to `/etc/profile` so all users have their own TMPDIR for the minikube command:
```bash
alias minikube="mkdir -p /tmp/${USER} && TMPDIR=/tmp/${USER} /usr/local/bin/minikube"
```

**Comment by medyagh** [comment]
the least we could do is adding the username to the logfile

the source of this we wanted "minikbue status" not to make many temp files so we had aded one file per command
so that caused multiple users coliding..

**Comment by andriyDev** [comment]
Fixed in #11294.
